src/main/java/gg/projecteden/nexus/features/shops/ResourceWorldCommand.java

```diff
@@ -0,0 +1,491 @@
+		if (selection.getArea() > 1000000)
+			error("Max selection size is 1000000");
+
+		for (Block block : worldEditUtils.getBlocks(selection))
+			logger.add(block.getLocation());
+
+		save(world());
+		send(PREFIX + "Added &e" + selection.getArea() + " &3locations to logger, new size: &e" + logger.size());
+	}
+
+	@Async
+	@Path("logger count")
+	@Permission("group.admin")
+	void logger_count() {
+		send(PREFIX + logger.size() + " coordinates logged");
+	}
+
+	@Async
+	@Environments(Env.TEST)
+	@Path("logger add random [amount]")
+	@Permission("group.admin")
+	void logger_add_random(@Arg("10000") int amount) {
+		if (isResourceWorld(world()))
+			throw new InvalidInputException("You must be in a resource world");
+
+		for (int i = 0; i < amount; i++) {
+			while (true) {
+				final Location location = getRandomLocation();
+				if (logger.contains(location))
+					continue;
+
+				logger.add(location);
+				break;
+			}
+		}
+
+		save(world());
+		send(PREFIX + "Added &e" + amount + " &3locations to logger, new size: &e" + getLogger(world()).size());
+	}
+
+	@NotNull
+	private Location getRandomLocation() {
+		final int x = RandomUtils.randomInt(-RADIUS, RADIUS);
+		final int y = RandomUtils.randomInt(-64, 319);
+		final int z = RandomUtils.randomInt(-RADIUS, RADIUS);
+		return new Location(world(), x, y, z);
+	}
+
+	@EventHandler
+	public void onEnterResourceWorld(PlayerTeleportEvent event) {
+		Player player = event.getPlayer();
+
+		if (!isResourceWorld(event.getTo()))
+			return;
+
+		if (isResourceWorld(event.getFrom()))
+			return;
+
+		if (Rank.of(player).isStaff())
+			return;
+
+		if (event.isCancelled())
+			return;
+
+		PlayerUtils.send(player, " &4Warning |");
+		PlayerUtils.send(player, " &4Warning | &cYou are entering the resource world!");
+		PlayerUtils.send(player, " &4Warning | &cThis world is regenerated on the &c&lfirst of every month&c,");
+		PlayerUtils.send(player, " &4Warning | &cso don't leave your stuff here or you will lose it!");
+		PlayerUtils.send(player, " &4Warning |");
+		PlayerUtils.send(player, " &4Warning | &cThe darkness is dangerous in this world");
+		PlayerUtils.send(player, " &4Warning |");
+	}
+
+	@EventHandler
+	public void onHomeBlockPlace(BlockPlaceEvent event) {
+		if (!isResourceWorld(event.getPlayer())) return;
+
+		List<Material> materials = new ArrayList<>(Arrays.asList(Material.CHEST, Material.TRAPPED_CHEST, Material.FURNACE, Material.BARREL));
+		materials.addAll(MaterialTag.WOODEN_DOORS.getValues());
+		if (!materials.contains(event.getBlockPlaced().getType()))
+			return;
+
+		Tip tip = new TipService().get(event.getPlayer());
+		if (tip.show(TipType.RESOURCE_WORLD_STORAGE))
+			PlayerUtils.send(event.getPlayer(), " &4Warning: &cYou are currently building in the resource world! " +
+				"This world is regenerated on the &c&lfirst of every month, &cso don't leave your stuff here or you will lose it!");
+	}
+
+	@EventHandler
+	public void onBlockPlace(BlockPlaceEvent event) {
```

Comment:
should these methods only log blocks sellable on the market?

Reply:
No, in case we add blocks to the market, and because there's no reason not to log all of them just to be safe